src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -95,5 +255,33 @@ public IObservable<string> GetPullRequestTemplate(ILocalRepositoryModel reposito
             return ret;
         }
 
+        void EnsurePullRefSpecExists(IRepository repo)
+        {
+            var spec = "+refs/pull/*/head:refs/remotes/origin/pr/*";
+            var origin = repo.Network.Remotes["origin"];
+            var existing = origin.FetchRefSpecs.FirstOrDefault(x => x.Specification == spec);
+
+            if (existing == null)
+            {
+                repo.Network.Remotes.Update(origin, x => x.FetchRefSpecs.Add(spec));
+            }
+        }
+
+        static string GetSafeBranchName(string name)
```

Comment:
The branch name that is created here is based on the PR title, so I think that carrying that through into the branch name wouldn't be desired. For example, if I create a PR called "Fix foo/bar errors" then it doesn't make much sense for the PR to be called `pr/123-fix foo/bar errors` especially as Visual Studio will display it in the "Manage Branches" window as:

- pr
  - 123-fix foo
    - bar errors